09d7a53d6506204611ea53affda7f5a33d052494
 APDPlatLocalContainerEntityManagerFactoryBean.java | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git APDPlatLocalContainerEntityManagerFactoryBean.java APDPlatLocalContainerEntityManagerFactoryBean.java
index 9c5644f..878befe 100644
--- APDPlatLocalContainerEntityManagerFactoryBean.java
+++ APDPlatLocalContainerEntityManagerFactoryBean.java
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package org.springframework.orm.jpa;
+package com.apdplat.platform.spring;
 
 import javax.persistence.EntityManagerFactory;
 import javax.persistence.PersistenceException;
@@ -28,6 +28,10 @@ import org.springframework.context.weaving.LoadTimeWeaverAware;
 import org.springframework.core.io.ResourceLoader;
 import org.springframework.instrument.classloading.LoadTimeWeaver;
 import org.springframework.jdbc.datasource.lookup.SingleDataSourceLookup;
+import org.springframework.orm.jpa.AbstractEntityManagerFactoryBean;
+import org.springframework.orm.jpa.EntityManagerFactoryInfo;
+import org.springframework.orm.jpa.JpaVendorAdapter;
+import org.springframework.orm.jpa.LocalEntityManagerFactoryBean;
 import org.springframework.orm.jpa.persistenceunit.DefaultPersistenceUnitManager;
 import org.springframework.orm.jpa.persistenceunit.PersistenceUnitManager;
 import org.springframework.orm.jpa.persistenceunit.PersistenceUnitPostProcessor;
@@ -78,13 +82,13 @@ import org.springframework.util.ClassUtils;
  * @see org.springframework.orm.jpa.support.SharedEntityManagerBean
  * @see javax.persistence.spi.PersistenceProvider#createContainerEntityManagerFactory
  */
-public class LocalContainerEntityManagerFactoryBean extends AbstractEntityManagerFactoryBean
+public class APDPlatLocalContainerEntityManagerFactoryBean extends AbstractEntityManagerFactoryBean
 		implements ResourceLoaderAware, LoadTimeWeaverAware {
 
 	private PersistenceUnitManager persistenceUnitManager;
 
-	private final DefaultPersistenceUnitManager internalPersistenceUnitManager =
-			new DefaultPersistenceUnitManager();
+	private final APDPlatPersistenceUnitManager internalPersistenceUnitManager =
+			new APDPlatPersistenceUnitManager();
 
 	private PersistenceUnitInfo persistenceUnitInfo;
 
@@ -116,7 +120,7 @@ public class LocalContainerEntityManagerFactoryBean extends AbstractEntityManage
 	 * <p><b>NOTE: Only applied if no external PersistenceUnitManager specified.</b>
 	 * @param persistenceXmlLocation a Spring resource String
 	 * identifying the location of the <code>persistence.xml</code> file
-	 * that this LocalContainerEntityManagerFactoryBean should parse
+	 * that this APDPlatLocalContainerEntityManagerFactoryBean should parse
 	 * @see #setPersistenceUnitManager
 	 */
 	public void setPersistenceXmlLocation(String persistenceXmlLocation) {
